src/gui/properties/speedplotview.cpp

```diff
@@ -350,31 +361,40 @@ void SpeedPlotView::paintEvent(QPaintEvent *)
     painter.setRenderHints(QPainter::Antialiasing);
 
     // draw graphs
-    rect.adjust(3, 0, 0, 0); // Need, else graphs cross left gridline
+    // averager is duration based, it may go little above the maxDuration
+    painter.setClipping(true);
+    painter.setClipRect(rect);
 
-    const double yMultiplier = (niceScale.arg == 0.0) ? 0.0 : (static_cast<double>(rect.height()) / niceScale.sizeInBytes());
-    const double xTickSize = static_cast<double>(rect.width()) / (m_viewablePointsCount - 1);
+    const DataCircularBuffer &queue = currentData();
 
-    boost::circular_buffer<PointData> &queue = getCurrentData();
+    // last point will be drawn at x=0, so we don't need it in the calculation of xTickSize
+    const milliseconds lastDuration {queue.empty() ? milliseconds {} : queue.back().duration};
+    const double xTickSize = static_cast<double>(rect.width()) / (m_currentMaxDuration - lastDuration).count();
+    const double yMultiplier = (niceScale.arg == 0.0) ? 0.0 : (static_cast<double>(rect.height()) / niceScale.sizeInBytes());
```

Comment:
you can use `{}` initializer for both double vars.

Reply:
>you can use `{}` initializer for both double vars.

Isn't it time to discuss our preferred initialization method?
It looks like we already have one approved aspect (we prefer to use braces instead of parentheses to initialize classes, although I think there may be exceptions).
But personally, I feel it's acceptable to use assignment for initialization. Even more than that, it looks more natural to me in many cases, especially when it comes to some numeric types (e.g. `int i = 10;` instead of `int i {10};`).

Anyway I don't mind if we adopt some stricter rules.

Reply:
>It looks like we already have one approved aspect (we prefer to use braces instead of parentheses to initialize classes, although I think there may be exceptions).

Agree.

>it looks more natural to me in many cases, especially when it comes to some numeric types (e.g. int i = 10; instead of int i {10};).

Perhaps we could agree on using assignment in initialization expressions when it comes to POD types.

Reply:
> it looks more natural to me in many cases, especially when it comes to some numeric types (e.g. int i = 10; instead of int i {10};).
>Perhaps we could agree on using assignment in initialization expressions when it comes to POD types.

why do guys insist on such rigid coding guidelines? I still haven't worked in a real company for programming. Do companies follow such rigid and explicit coding guidelines?

Reply:
> I still haven't worked in a real company for programming. Do companies follow such rigid and explicit coding guidelines?

Programming is not my main profession. I use it as an auxiliary tool in my life, and as a hobby. But for the sake of interest, you can take a look at [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html). They are quite extensive.